### ISSUE TITLE
60421754 : [React Native Gallery App New]: Keyboard focus temporarily disappears after collapsing the navigation panel.

### ISSUE THREAD

**Issue by vineethkuttan** [microsoft/react-native-gallery#788]
## ADO Bug Reference

| Field | Value |
|-------|-------|
| **ADO Link** | 60421754 |
| **Type** | Bug |
| **State** | Active |
| **Priority** | -- |
| **Severity** | -- |
| **Reason** | Investigate |
| **Area Path** | `--` |
| **Iteration** | `--` |
| **Created By** | Arjun Chakraborty (WIPRO LIMITED) |
| **Assigned To** | Nitin Chaudhary |

---

### Environment Details

OS Version: 24H2
OS Build: 27975.984
Application: React Native Gallery
Application Version: 2.0.8.0
AT: Keyboard

### Repro Steps

1. Open the application and sign in with valid credentials.
2. Navigate to 'Navigation menu' button and invoke it and observe the keyboard focus.
### Actual Results

When the user collapses the navigation panel, the keyboard focus is lost for one interaction. The focus does not appear on any visible element immediately after collapsing, and it becomes visible again only after the user presses Tab once more.

### Screen Reader Behavior

After the navigation is collapsed, the screen reader does not announce any newly focused element because the focus is not correctly set. Just it's announcing the success info as the navigation menu has collapsed.

### Expected Results

Collapsing the navigation should not cause the keyboard focus to disappear. The focus should shift to a meaningful and visible element immediately after the action, ensuring continuity in the navigation flow.

### User Impact

Keyboard-only users may become disoriented because the focus temporarily disappears, breaking the navigation sequence. This can cause confusion, slow task completion, and create barriers for users who rely on consistent focus visibility, including users with motor or visual impairments.

### MAS Reference

[MAS 2.4.7 – Focus Visible](https://aka.ms/MAS2.4.7)

### Trap ID

[2.1 - Physical Challenge](https://aka.ms/PhysicalChallenge)

**“Have feedback on bugs logged, please tag bug as “A11yRCA” and add your feedback in the comment section.”**

---

> _Automatically created from [ADO Bug 60421754](#) using ADO to GitHub extension_

**Comment by vineethkuttan** [comment]
Checked copilot and Winui3 gallery, they also have the same behaviour, checking with grading team.

<img width="714" height="1037" alt="Image" src="https://github.com/user-attachments/assets/2944a8a0-e46f-4d85-96a1-8a7529dd15f5" />
